src/reltoaltrep.cpp

```diff
@@ -261,13 +273,15 @@ static R_altrep_class_t LogicalTypeToAltrepType(const LogicalType &type) {
 	case LogicalTypeId::VARCHAR:
 	case LogicalTypeId::UUID:
 		return RelToAltrep::string_class;
+	case LogicalTypeId::LIST:
+		return RelToAltrep::list_class;
 	default:
 		cpp11::stop("rel_to_altrep: Unknown column type for altrep: %s", type.ToString().c_str());
 	}
 }
 
 [[cpp11::register]] SEXP rapi_rel_to_altrep(duckdb::rel_extptr_t rel) {
-	D_ASSERT(rel && rel->rel);
+  	D_ASSERT(rel && rel->rel);
```

Comment:
Can you please clean this?

Reply:
Will do. I just need to make the use case I started from independent of `duckplyr`, in `test-list.R` ? 

``` r
library(duckdb)
library(duckplyr)

df <- tibble(a = list(1, c(1,2))) |>
  as_duckplyr_df()

filter(df, 0 == 0)
#> materializing:
#> ---------------------
#> --- Relation Tree ---
#> ---------------------
#> Filter [==(0.0, 0.0)]
#>   r_dataframe_scan(0x12b4a5e30)
#> 
#> ---------------------
#> -- Result Columns  --
#> ---------------------
#> - a (DOUBLE[])
#> 
#> # A tibble: 2 × 1
#>   a        
#>   <list>   
#> 1 <dbl [1]>
#> 2 <dbl [2]>
```

<sup>Created on 2024-02-22 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>